src/main/java/mate/academy/repository/BookRepositoryImpl.java

```diff
@@ -33,4 +32,13 @@ public List<Book> findAll() {
             throw new DataProcessingException("Fetching books failed", e);
         }
     }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try {
+            return sessionFactory.fromSession(s -> Optional.ofNullable(s.get(Book.class, id)));
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't find book by id: " + id, e);
```

Comment:
Maybe it's not very good exception message in this case, because you have the same message in BookService class.